src/js/select2/results.js

```diff
@@ -14,7 +14,7 @@ define([
 
   Results.prototype.render = function () {
     var $results = $(
-      '<ul class="select2-results__options" role="tree"></ul>'
+      '<ul class="select2-results__options" role="listbox"></ul>'
```

Comment:
How well does this work with a nested [`group` role](https://www.w3.org/TR/wai-aria/roles#group)?

Part of the reason why `tree` was picked was because it supported nesting of options. From what we could tell, `listbox` only supports a single level (completely flat).

Reply:
> I'm afraid ARIA doesn't provide a way to emulate the semantics of optgroup
Also note that plain old HTML optgroups don't get conveyed to screen readers either.

Reply:
For reference, in ARIA 1.2 `role=group` will be allowed as child of `role=listbox` to emulate `optgroup`. See https://github.com/w3c/aria/pull/782